soccer/KickEvaluator.cpp

```diff
@@ -0,0 +1,396 @@
+#include "KickEvaluator.hpp"
+#include <Utils.hpp>
+#include <Geometry2d/Util.hpp>
+
+#include <algorithm>
+#include <vector>
+#include <math.h>
+#include <cmath>
+
+static union {
+    float d;
+    struct {
+#ifdef LITTLE_ENDIAN
+        int j, i;
+#else
+        int i, j;
+#endif
+    } n;
+} _eco;
+
+// Floating point magic
+#define EXP_A (1048576 / 0.69314718055994530942)
+#define EXP_C 60801
+#define fast_exp(y) (_eco.n.i = EXP_A * (y) + (1072693248 - EXP_C), _eco.d)
```

Comment:
Bad news, at -O2, the Taylors Series is still twice as slow. Good news, I can now go through half a million executions in only 2.3 seconds.